### PR TITLE
cmd: add new "snap recovery" command

### PR DESCRIPTION
**Pull request by mvo5**
The new "snap recovery" command lists the available recovery
systems.

### CODE REVIEW

File: cmd/snap/cmd_recovery.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	w := tabWriter()
+	defer w.Flush()
+	fmt.Fprintf(w, i18n.G("Label\tBrand\tModel\tNotes\n"))
+	for _, sys := range systems {
+		// doing it this way because otherwise it's a sea of %s\t%s\t%s
+		line := []string{
+			sys.Label,
+			sys.Brand.DisplayName,
+			sys.Model.DisplayName,
```
Comment (pedronis) [inline cmd/snap/cmd_recovery.go:76]: I'm not sure we should use the DisplayName of the model here, for the same reasons that "snap list" just uses the name, not the title or something else for snaps

File: cmd/snap/cmd_recovery.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	}
+
+	w := tabWriter()
+	defer w.Flush()
+	fmt.Fprintf(w, i18n.G("Label\tBrand\tModel\tNotes\n"))
+	for _, sys := range systems {
+		// doing it this way because otherwise it's a sea of %s\t%s\t%s
+		line := []string{
+			sys.Label,
+			sys.Brand.DisplayName,
```
Comment (pedronis) [inline cmd/snap/cmd_recovery.go:75]: we need to use something like shortPublisher here like we do in "snap list"

File: cmd/snap/cmd_recovery.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+)
+
+type cmdRecovery struct {
+	clientMixin
+}
```
Comment (mvo5) [inline cmd/snap/cmd_recovery.go:34]: Note from Samuele: `if we want more/nicer details  we can probably  add later "snap recovery <system>"` - there should be a XXX for this :)

File: cmd/snap/cmd_recovery.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	systems, err := x.client.ListSystems()
+	if err != nil {
+		return fmt.Errorf("cannot list recovery systems: %v", err)
+	}
+	if len(systems) == 0 {
+		fmt.Fprintf(Stdout, "no recovery systems found\n")
```
Comment (bboozzoo) [inline cmd/snap/cmd_recovery.go:65]: Should this rather go to stderr?

File: cmd/snap/cmd_recovery.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+	esc := x.getEscapes()
+	w := tabWriter()
+	defer w.Flush()
+	fmt.Fprintf(w, i18n.G("Label\tBrand\tModel\tNotes\n"))
+	for _, sys := range systems {
+		// doing it this way because otherwise it's a sea of %s\t%s\t%s
+		line := []string{
+			sys.Label,
+			shortPublisher(esc, &sys.Brand),
```
Comment (bboozzoo) [inline cmd/snap/cmd_recovery.go:77]: This also prints out a tick mark when the account is a verified publisher. Should we keep this behavior?

File: cmd/snap/cmd_recovery.go (hunk 4 above)
Comment (pedronis) [reply cmd/snap/cmd_recovery.go:65]: yes, we should follow what "snap list" does for that.

File: cmd/snap/cmd_recovery.go (hunk 5 above)
Comment (pedronis) [reply cmd/snap/cmd_recovery.go:77]: we use the publisher methods with that behavior also for snap model, so I would say yes

Comment (mvo5) [reply cmd/snap/cmd_recovery.go:77]: Yeah, I think so too.

Comment (mvo5) [reply cmd/snap/cmd_recovery.go:77]: So no change needed here or am I misreading this?

File: cmd/snap/cmd_recovery.go (hunk 4 above)
Comment (mvo5) [reply cmd/snap/cmd_recovery.go]: Fixed now.

File: cmd/snap/cmd_recovery.go (hunk 3 above)
Comment (mvo5) [reply cmd/snap/cmd_recovery.go:35]: Added this now too

File: cmd/snap/cmd_recovery.go (hunk 5 above)
Comment (pedronis) [reply cmd/snap/cmd_recovery.go:78]: yes, no change

**Pull request merged**